builders/gradle.go

```diff
@@ -139,10 +145,21 @@ func (builder *GradleBuilder) DiscoverModules(dir string) ([]module.Config, erro
 				}
 			}
 
-			return moduleConfigurations, nil
+			if len(moduleConfigurations) > 0 {
+				return moduleConfigurations, nil
+			}
+
+			// If task list succeeds but returns no subproject dependencies tasks, fall back to the root `dependencies` task
+			return []module.Config{
+				{
+					Name: ":compile",
+					Path: "build.gradle",
+					Type: "gradle",
+				},
+			}, nil
 		}
 
-		// Fall back to "app" as default task, even though technically it would be "" (root)
+		// If task list fails, fall back to "app" as default task, even though technically it would be "" (root)
```

Comment:
Is this behaviour correct? Should we expect `app:compile` to work even when `gradle tasks` fails, or should this be a fatal error?

Reply:
I'm 50/50 on having this fallback vs erroring if `gradle tasks` fails, but:

 1. I believe `app:compile` is a sane default if a `build.gradle` is found
 2. Checking for a `build.gradle` is a valid way of identifying a module, and much simpler/more reliable then parsing output of `gradle tasks`

Reply:
I think the question is whether `app:compile` is likely to work. If it works, then this is great. If it doesn't work, then we've kicked a configuration error down the line into a much more obscure runtime error.

I don't have much experience with the Gradle ecosystem, so I trust your intention on this.